src/marqo/core/search/recommender.py

```diff
@@ -21,52 +22,29 @@ def __init__(self, vespa_client: VespaClient, index_management: IndexManagement,
         self.index_management = index_management
         self.inference = inference
 
-    def recommend(self,
+    def get_doc_vectors_from_ids(self,
                   index_name: str,
                   documents: Union[List[str], Dict[str, float]],
                   tensor_fields: Optional[List[str]] = None,
-                  interpolation_method: Optional[InterpolationMethod] = None,
-                  exclude_input_documents: bool = True,
-                  result_count: int = 3,
-                  offset: int = 0,
-                  highlights: bool = True,
-                  ef_search: Optional[int] = None,
-                  approximate: Optional[bool] = None,
-                  searchable_attributes: Optional[List[str]] = None,
-                  verbose: int = 0,
-                  reranker: Union[str, Dict] = None,
-                  filter: str = None,
-                  attributes_to_retrieve: Optional[List[str]] = None,
-                  score_modifiers: Optional[ScoreModifierLists] = None,
-                  rerank_depth: Optional[int] = None
-                  ):
+                  concurrency: Optional[int] = None) -> Dict[str, List[List[float]]]:
         """
-        Recommend documents similar to the provided documents.
+        This method gets documents from Vespa using their IDs, removes any unnecessary data, checks for
+        lack of vectors, then returns a list of document vectors. Can be used internally (in recommend)
+        or externally (in the search module).
 
         Args:
             index_name: Name of the index to search
             documents: A list of document IDs or a dictionary where the keys are document IDs and the values are weights
             tensor_fields: List of tensor fields to use for recommendation (can include text, image, audio, and video fields)
-            interpolation_method: Interpolation method to use for combining vectors
-            exclude_input_documents: Whether to exclude the input documents from the search results
-            result_count: Number of results to return
-            offset: Offset of the first result
-            highlights: Whether to include highlights in the results
-            ef_search: ef_search parameter for HNSW search
-            approximate: Whether to use approximate search
-            searchable_attributes: List of attributes to search in
-            verbose: Verbosity level
-            reranker: Reranker to use
-            filter: Filter string
-            attributes_to_retrieve: List of attributes to retrieve
-            score_modifiers: Score modifiers to apply
-            rerank_depth: Rerank depth
+            concurrency: Max number of concurrent requests to use when fetching documents by batch
+
+        Returns:
+            A dictionary mapping document IDs to lists of vector embeddings
```

Comment:
nit: It would be better to be clear that this method returns the vectors of the corresponding tensor fields. Also, is the order of the vectors guaranteed here?

Reply:
Added to docstring: This is flattened to 1 list per document ID (not separated by tensor field). Order of embeddings is not guaranteed.